File: app/src/main/java/com/billpayments/features/home/ui/HomeScreenData.kt
package com.billpayments.features.home.ui

import androidx.compose.runtime.Immutable

@Immutable
data class CategoryItemData(val id: String, val name: String)

@Immutable
data class DueBillItemData(
    val billerId: String,
    val account: String,
    val billerName: String,
    val amountPaise: Long,
    val dueInDays: Int,
)

@Immutable
data class SavedBillerItemData(
    val billerId: String,
    val account: String,
    val nickname: String,
    val billerName: String,
    val openAmount: Boolean,
)

@Immutable
sealed interface HomeRemindersData {
    data object Loading : HomeRemindersData
    data class Loaded(val items: List<DueBillItemData>) : HomeRemindersData
}

@Immutable
sealed interface HomeSavedBillersData {
    data object Loading : HomeSavedBillersData
    data class Loaded(val items: List<SavedBillerItemData>) : HomeSavedBillersData
}

@Immutable
sealed interface HomeCategoriesData {
    data object Loading : HomeCategoriesData
    data class Error(val message: String) : HomeCategoriesData
    data class Loaded(val items: List<CategoryItemData>) : HomeCategoriesData
}

File: BillPayments/Features/Home/UI/HomeScreenData.swift
struct CategoryItemData: Equatable, Identifiable {
    let id: String
    let name: String
}

struct DueBillItemData: Equatable, Identifiable {
    let billerId: String
    let account: String
    let billerName: String
    let amountPaise: Int64
    let dueInDays: Int

    var id: String { "\(billerId):\(account)" }
}

struct SavedBillerItemData: Equatable, Identifiable {
    let billerId: String
    let account: String
    let nickname: String
    let billerName: String
    let openAmount: Bool

    var id: String { "\(billerId):\(account)" }
}

enum HomeRemindersData: Equatable {
    case loading
    case loaded([DueBillItemData])
}

enum HomeSavedBillersData: Equatable {
    case loading
    case loaded([SavedBillerItemData])
}

enum HomeCategoriesData: Equatable {
    case loading
    case error(String)
    case loaded([CategoryItemData])
}
